Source/inv.cpp

```diff
@@ -1942,13 +1942,13 @@ void ConsumeScroll(Player &player)
 	int itemSlot = player.executedSpell.spellFrom;
 	int itemIndex = 0;
 	Item *item;
-	if (itemSlot <= INVITEM_INV_LAST) {
-		itemIndex = itemSlot - INVITEM_INV_FIRST;
-		item = &player.InvList[itemIndex];
-	} else {
-		itemIndex = itemSlot - INVITEM_BELT_FIRST;
-		item = &player.SpdList[itemIndex];
-	}
+        if (itemSlot >= INVITEM_INV_FIRST && itemSlot <= INVITEM_INV_LAST) {
+                itemIndex = itemSlot - INVITEM_INV_FIRST;
+                item = &player.InvList[itemIndex];
+        } else if (itemSlot >= INVITEM_BELT_FIRST) {
+                itemIndex = itemSlot - INVITEM_BELT_FIRST;
+                item = &player.SpdList[itemIndex];
+        }

```

Comment:
else `app_fatal` because this should never happen

Reply:
Currently this can happen (0 is set) when using the spell list to cast a scroll.
Cause in this case no explicit scroll was selected/used.

Reply:
Ah, makes sense